File: mParticle-Apple-SDK/MPRoktEvent.m
#import "MPRoktEvent.h"

#pragma mark - MPRoktEvent

@implementation MPRoktEvent
@end

#pragma mark - MPRoktInitComplete

@interface MPRoktInitComplete ()
@property (nonatomic, readwrite) BOOL success;
@end

@implementation MPRoktInitComplete

- (instancetype)initWithSuccess:(BOOL)success {
    self = [super init];
    if (self) {
        _success = success;
    }
    return self;
}

@end

#pragma mark - MPRoktShowLoadingIndicator

@implementation MPRoktShowLoadingIndicator
@end

#pragma mark - MPRoktHideLoadingIndicator

@implementation MPRoktHideLoadingIndicator
@end

#pragma mark - MPRoktPlacementInteractive

@interface MPRoktPlacementInteractive ()
@property (nonatomic, readwrite, nullable) NSString *placementId;
@end

@implementation MPRoktPlacementInteractive

- (instancetype)initWithPlacementId:(nullable NSString *)placementId {
    self = [super init];
    if (self) {
        _placementId = placementId;
    }
    return self;
}

@end

#pragma mark - MPRoktPlacementReady

@interface MPRoktPlacementReady ()
@property (nonatomic, readwrite, nullable) NSString *placementId;
@end

@implementation MPRoktPlacementReady

- (instancetype)initWithPlacementId:(nullable NSString *)placementId {
    self = [super init];
    if (self) {
        _placementId = placementId;
    }
    return self;
}

@end

#pragma mark - MPRoktOfferEngagement

@interface MPRoktOfferEngagement ()
@property (nonatomic, readwrite, nullable) NSString *placementId;
@end

@implementation MPRoktOfferEngagement

- (instancetype)initWithPlacementId:(nullable NSString *)placementId {
    self = [super init];
    if (self) {
        _placementId = placementId;
    }
    return self;
}

@end

#pragma mark - MPRoktOpenUrl

@interface MPRoktOpenUrl ()
@property (nonatomic, readwrite, nullable) NSString *placementId;
@property (nonatomic, readwrite) NSString *url;
@end

@implementation MPRoktOpenUrl

- (instancetype)initWithPlacementId:(nullable NSString *)placementId url:(NSString *)url {
    self = [super init];
    if (self) {
        _placementId = placementId;
        _url = url;
    }
    return self;
}

@end

#pragma mark - MPRoktPositiveEngagement

@interface MPRoktPositiveEngagement ()
@property (nonatomic, readwrite, nullable) NSString *placementId;
@end

@implementation MPRoktPositiveEngagement

- (instancetype)initWithPlacementId:(nullable NSString *)placementId {
    self = [super init];
    if (self) {
        _placementId = placementId;
    }
    return self;
}

@end

#pragma mark - MPRoktPlacementClosed

@interface MPRoktPlacementClosed ()
@property (nonatomic, readwrite, nullable) NSString *placementId;
@end

@implementation MPRoktPlacementClosed

- (instancetype)initWithPlacementId:(nullable NSString *)placementId {
    self = [super init];
    if (self) {
        _placementId = placementId;
    }
    return self;
}

@end

#pragma mark - MPRoktPlacementCompleted

@interface MPRoktPlacementCompleted ()
@property (nonatomic, readwrite, nullable) NSString *placementId;
@end

@implementation MPRoktPlacementCompleted

- (instancetype)initWithPlacementId:(nullable NSString *)placementId {
    self = [super init];
    if (self) {
        _placementId = placementId;
    }
    return self;
}

@end

#pragma mark - MPRoktPlacementFailure

@interface MPRoktPlacementFailure ()
@property (nonatomic, readwrite, nullable) NSString *placementId;
@end

@implementation MPRoktPlacementFailure

- (instancetype)initWithPlacementId:(nullable NSString *)placementId {
    self = [super init];
    if (self) {
        _placementId = placementId;
    }
    return self;
}

@end

#pragma mark - MPRoktFirstPositiveEngagement

@interface MPRoktFirstPositiveEngagement ()
@property (nonatomic, readwrite, nullable) NSString *placementId;
@end

@implementation MPRoktFirstPositiveEngagement

- (instancetype)initWithPlacementId:(nullable NSString *)placementId {
    self = [super init];
    if (self) {
        _placementId = placementId;
    }
    return self;
}

@end

#pragma mark - MPRoktCartItemInstantPurchase

@interface MPRoktCartItemInstantPurchase ()
@property (nonatomic, readwrite) NSString *placementId;
@property (nonatomic, readwrite, nullable) NSString *name;
@property (nonatomic, readwrite) NSString *cartItemId;
@property (nonatomic, readwrite) NSString *catalogItemId;
@property (nonatomic, readwrite) NSString *currency;
@property (nonatomic, readwrite) NSString *itemDescription;
@property (nonatomic, readwrite, nullable) NSString *linkedProductId;
@property (nonatomic, readwrite) NSString *providerData;
@property (nonatomic, readwrite, nullable) NSDecimalNumber *quantity;
@property (nonatomic, readwrite, nullable) NSDecimalNumber *totalPrice;
@property (nonatomic, readwrite, nullable) NSDecimalNumber *unitPrice;
@end

@implementation MPRoktCartItemInstantPurchase

- (instancetype)initWithPlacementId:(NSString *)placementId
                               name:(nullable NSString *)name
                         cartItemId:(NSString *)cartItemId
                      catalogItemId:(NSString *)catalogItemId
                           currency:(NSString *)currency
                        description:(NSString *)description
                    linkedProductId:(nullable NSString *)linkedProductId
                       providerData:(NSString *)providerData
                           quantity:(nullable NSDecimalNumber *)quantity
                         totalPrice:(nullable NSDecimalNumber *)totalPrice
                          unitPrice:(nullable NSDecimalNumber *)unitPrice {
    self = [super init];
    if (self) {
        _placementId = placementId;
        _name = name;
        _cartItemId = cartItemId;
        _catalogItemId = catalogItemId;
        _currency = currency;
        _itemDescription = description;
        _linkedProductId = linkedProductId;
        _providerData = providerData;
        _quantity = quantity;
        _totalPrice = totalPrice;
        _unitPrice = unitPrice;
    }
    return self;
}

- (NSString *)description {
    return _itemDescription;
}

@end

File: mParticle-Apple-SDK/MPRoktEvent.swift
//
//  MPRoktEvent.swift
//  mParticle-Apple-SDK
//
//  Created by Thomson Thomas on 6/17/25.
//
//  This file contains the Swift classes for the MPRoktEvent enum.
//

import Foundation
import UIKit

@objc public class MPRoktEvent: NSObject {
    @objc public class MPRoktInitComplete: MPRoktEvent {
        @objc public let success: Bool

        @objc public init(success: Bool) {
            self.success = success
            super.init()
        }
    }

    @objc public class MPRoktShowLoadingIndicator: MPRoktEvent {
        @objc override public init() {
            super.init()
        }
    }

    @objc public class MPRoktHideLoadingIndicator: MPRoktEvent {
        @objc override public init() {
            super.init()
        }
    }

    @objc public class MPRoktPlacementInteractive: MPRoktEvent {
        @objc public let placementId: String?

        @objc public init(placementId: String?) {
            self.placementId = placementId
            super.init()
        }
    }

    @objc public class MPRoktPlacementReady: MPRoktEvent {
        @objc public let placementId: String?

        @objc public init(placementId: String?) {
            self.placementId = placementId
            super.init()
        }
    }

    @objc public class MPRoktOfferEngagement: MPRoktEvent {
        @objc public let placementId: String?

        @objc public init(placementId: String?) {
            self.placementId = placementId
            super.init()
        }
    }

    @objc public class MPRoktOpenUrl: MPRoktEvent {
        @objc public let placementId: String?
        @objc public let url: String

        @objc public init(placementId: String?, url: String) {
            self.url = url
            self.placementId = placementId
            super.init()
        }
    }

    @objc public class MPRoktPositiveEngagement: MPRoktEvent {
        @objc public let placementId: String?

        @objc public init(placementId: String?) {
            self.placementId = placementId
            super.init()
        }
    }

    @objc public class MPRoktPlacementClosed: MPRoktEvent {
        @objc public let placementId: String?

        @objc public init(placementId: String?) {
            self.placementId = placementId
            super.init()
        }
    }

    @objc public class MPRoktPlacementCompleted: MPRoktEvent {
        @objc public let placementId: String?

        @objc public init(placementId: String?) {
            self.placementId = placementId
            super.init()
        }
    }

    @objc public class MPRoktPlacementFailure: MPRoktEvent {
        @objc public let placementId: String?

        @objc public init(placementId: String?) {
            self.placementId = placementId
            super.init()
        }
    }

    @objc public class MPRoktFirstPositiveEngagement: MPRoktEvent {
        @objc public let placementId: String?

        @objc public init(placementId: String?) {
            self.placementId = placementId
            super.init()
        }
    }

    @objc public class MPRoktCartItemInstantPurchase: MPRoktEvent {
        @objc public let placementId: String
        @objc public let name: String?
        @objc public let cartItemId: String
        @objc public let catalogItemId: String
        @objc public let currency: String
        private let _description: String
        @objc override public var description: String {
            _description
        }

        @objc public let linkedProductId: String?
        @objc public let providerData: String
        @objc public let quantity: NSDecimalNumber?
        @objc public let totalPrice: NSDecimalNumber?
        @objc public let unitPrice: NSDecimalNumber?

        @objc public init(placementId: String,
                          name: String?,
                          cartItemId: String,
                          catalogItemId: String,
                          currency: String,
                          description: String,
                          linkedProductId: String?,
                          providerData: String,
                          quantity: NSDecimalNumber?,
                          totalPrice: NSDecimalNumber?,
                          unitPrice: NSDecimalNumber?) {
            self.placementId = placementId
            self.name = name
            self.cartItemId = cartItemId
            self.catalogItemId = catalogItemId
            self.currency = currency
            _description = description
            self.linkedProductId = linkedProductId
            self.providerData = providerData
            self.quantity = quantity
            self.totalPrice = totalPrice
            self.unitPrice = unitPrice
            super.init()
        }
    }

    /// An event that is emitted when the height of an embedded placement changes.
    /// This event is useful for dynamically adjusting the container view's height
    /// to accommodate the Rokt placement content.
    ///
    /// - Note: This event is only emitted for embedded placements
    @objc public class MPRoktEmbeddedSizeChanged: MPRoktEvent {
        /// The identifier of the placement whose height has changed
        @objc public let placementId: String
        /// The new height of the placement
        @objc public let updatedHeight: CGFloat

        @objc public init(placementId: String, updatedHeight: CGFloat) {
            self.placementId = placementId
            self.updatedHeight = updatedHeight
            super.init()
        }
    }
}
